### ISSUE TITLE
Implement HomeScreen.md template

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#496]
The [HomeScreen.md](https://github.com/phetsims/joist/blob/master/doc/HomeScreen.md) file is not just for the homescreen, but outlines how the entire joist scaffolding should look in the pDOM. Currently, as part of https://github.com/phetsims/area-model-common/issues/149, the headers are working, but we need to do the rest of this.

**Comment by zepumph** [comment]
As already noted, general, cross sim commits regarding screen headers were made in https://github.com/phetsims/area-model-common/issues/149. @jessegreenberg please review these commits:
https://github.com/phetsims/joist/commit/481c860064c9b9e5f866f837105064b61181ac3d
https://github.com/phetsims/joist/commit/6588aeacac4ed83f32a864aab3c5bd2255cce653
https://github.com/phetsims/joist/commit/3781c9037198222c87921798706faaecbd64835f
https://github.com/phetsims/balloons-and-static-electricity/commit/2ffdedd4ac18bbe405dabdfd21096f093efa5fe0
https://github.com/phetsims/resistance-in-a-wire/commit/e2ad76cbdeb1d496821b210323395bcd7ed0b63e
https://github.com/phetsims/ohms-law/commit/0b55802d3e5a704f145b12ed611eefe4af55f4fa

I see one potential issue with my implementation. If there is a screen name, but the sim is a single screen sim, like for area model decimals, whose only screen is sometimes used in other, multiscreen sims, so its title looks like "Decimals‬ Screen, ‪Area Model Decimals‬"

I could be ok with that because it is only more information, but it is inconsistent.

**Comment by zepumph** [comment]
Tagging @terracoda too to note issues she sees about joist related strings over in this issue.

**Comment by jessegreenberg** [comment]
Hmm, interesting case. Then maybe we need to make the view's `labelContent` truly dependent on the number of screens rather than existence of sim name. One way to do this would be to create the `labelContent` in `Sim.start` where we can count the number of screens. But I really like that the implementation is in Screen.js rather than Sim.js.

Here is another option for your consideration, I am not sure I love it yet:
`Screen.initializeView` takes `options` rather than an optional `simName`, and https://github.com/phetsims/joist/blob/master/js/Screen.js#L267-L286 could change to 
```
      // If a single screen sim, the accessible label for the view is just the sim name - otherwise it is the
      // screen and sim name comma separated like "Simulation Screen, Simulation Name"
      if ( options.simName ) {
        if ( options.numberOfScreens === 1 ) {
          this._view.labelContent = options.simName;
        }
        else {

          // Like "My Awesome Screen" because "My Awesome" is the name of the screen.
          var screenNameWithScreen = StringUtils.fillIn( screenNamePatternString, {
            name: this.name
          } );

          // initialize proper pDOM labelling for ScreenView
          this._view.labelContent = StringUtils.fillIn( screenSimPatternString, {
            screenName: screenNameWithScreen,
            simName: options.simName
          } );
        }
      }
```

**Comment by zepumph** [comment]
I don't want to let this sit in master, high priority.

**Comment by zepumph** [comment]
@jessegreenberg good idea! How about that?

**Comment by terracoda** [comment]
I think it makes sense to make it dependent on the number of screens. That makes sense to me and is in line with how I created the HomeScreen pdom content.

**Comment by terracoda** [comment]
@zepumph, FYI, we just reviewed the addition of the global mute sound button in the Design Meeting, and decided that it needs to be on the Home Scree, so both the global mute sound button and the keyboard shortcuts button will be present on the Home Screen.

We'll  figure out the content of the Home Screen shortcuts dialog separately.

**Comment by zepumph** [comment]
@jessegreenberg please review https://github.com/phetsims/joist/commit/80a44cefe8c4cd1ba2013b5630d7f801f39b4240.

**Comment by jessegreenberg** [comment]
Sorry @zepumph, I just reviewed the above commit it looks very nice. I also reviewed a few single and multi-screen sims and the screen strings look correct. Is anything remaining for this issue?

**Comment by zepumph** [comment]
The first thing that I notice is that `HomeScreen.js` doesn't have a name, so it comes up as "null Screen, Coulombs Law" as the h1.  I'll start there.

**Comment by zepumph** [comment]
Implemented above. I looked around to see if it was scary to pass that name in, but couldn't find anything so I committed.

**Comment by zepumph** [comment]
`HomeScreen.md` has an aria-labelledby between the `nav` container and `Sim Screens` as an `h2`. In master it was an `aria-label` and there was no `h2`. I changed than above.

**Comment by zepumph** [comment]
In the above commits I removed the "Play Area" and "Control Area" sections from the home screen.

**Comment by zepumph** [comment]
* Something that is in master but not in the `HomeScreen.md` example is `aria-roledescription="Sim Screen"` on the `button` for each screen `li`. Should we keep that attribute on each screen `button`?

**Comment by pixelzoom** [comment]
@zepumph asked in Slack:

> 
Michael Kauzmann:octopus:  7:17 PM
Can anyone think of a reason why HomeScreen.js should NOT have a name provided (like if something assumes it will have a name of null)? If so please comment in https://github.com/phetsims/joist/issues/496 where I passed a name in to `
HomeScreen.js.

I'm not aware of anything that is relying on HomeScreen having a null name. And if something is doing that, if should be identified and changed.

**Comment by zepumph** [comment]
I added the aria-describedby association between the `button` and the `p` for each screen in the above commit.

**Comment by zepumph** [comment]
https://github.com/phetsims/joist/issues/496#issuecomment-603023804 was incomplete. We also wanted to remove the default screen summary Node. Implemented a more general solution above.

**Comment by zepumph** [comment]
@terracoda this issue is ready for preliminary review. Here is what I did:

* I looked at `HomeScreen.md` and the example there of Coulomb's Law Home Screen.
* I then compared that to what is on master for the CL home screen.
* I tried to be as clear as possible above as to what I changed to align these two.

What I think is best for you to review:
* Take a look at `HomeScreen.md` to refresh (this issue is 2 years old), especially the CL code snippet. For this particular review I have only made changes to the HomeScreen, and think it would be best to limit the conversation to just that screen. Please make other issues if you see problems outside of the home screen.
* Look at each comment/commit pair since https://github.com/phetsims/joist/issues/496#issuecomment-602939588 and make sure that these changes are as expected.
* Use Coulombs Law and Greenhouse Effect in the a11y view as examples of the PDOM of the home screen. 
* Please note that each screen's "description" p has not been implemented yet, and thus is just the default: `{{screenName}} Screen`. There is logic set up to be able to change that.

Good luck! Let me know if you want to meet or anything. The priority of this issue is tied to the progress of interactive descriptions on greenhouse effect, as it will be the first multiscreen sim we outfit with interactive descriptions.

**Comment by zepumph** [comment]
@emily-phet, I think it would be good to ping this issue as common code work needed to be done before publishing our first multi-screen simulation with interactive description (ration and proportion). Please prioritize accordingly.

**Comment by emily-phet** [comment]
Let's make a plan for this tomorrow.

**Comment by terracoda** [comment]
in https://github.com/phetsims/joist/issues/496#issuecomment-603027374, @zepumph asked:
>Something that is in master but not in the HomeScreen.md example is aria-roledescription="Sim Screen" on the button for each screen li. Should we keep that attribute on each screen button?

We discussed the use of `aria-roledescription` in the initial design and I think we decided not to use it, or at least not to get rid of the word "button". People generally felt  that it could be confusing to rename a native role like "button" which is so intuitive.

If we use `aria-roledescription`, the string needs to be, "sim screen button", and not "sim screen" alone. And we should only use it if we think and we notice in interviews that it enhances the user experience. See [advice from Marco Zehe](https://www.marcozehe.de/easy-aria-tip-8-use-aria-roledescription-to-selectively-enhance-the-user-experience/).

If there is something special or different about the sim screen buttons, AND we want to communicate that difference, we should re-consider using the attribute with "sim screen button".

**Comment by terracoda** [comment]
re comment https://github.com/phetsims/joist/issues/496#issuecomment-603022191
When a section or region of some kind has a heading associated with `aria-labelledby`, I don't thin it should need a an `aria-label`.

**Comment by terracoda** [comment]
@zepumph, is it okay to use RaP as the example to review? It's the latest example.

**Comment by terracoda** [comment]
![Screen Shot 2020-07-16 at 12 11 45 PM](https://user-images.githubusercontent.com/1902116/87685526-14eab200-c75e-11ea-8f88-02ac56f82dad.png)

Remove "Home Screen", this is the landing page for the sim, so I think it is best to just name the sim, like this:
```html
<!-- Home Screen -->
<h1>{{SimName}}</h1> 
  <p>{{GeneralActionPhrase}} {{SimName}}. It has {{NumScreens}} interactive screens.</p>	
```

If others disagree, I would flip it, so "home screen" came after the sim name like this:
```html
<!-- Home Screen -->
<h1>{{SimName}}, home screen</h1> 
  <p>{{GeneralActionPhrase}} {{SimName}}. It has {{NumScreens}} interactive screens.</p>	
```

But my preference is no "home screen" string.

As mentioned in #641, I think `aria-describedby` would be nice on the home screen.

**Comment by terracoda** [comment]
@zepumph said
> I added the `aria-describedby` association between the button and the p for each screen in the above commit.

I think it is better to have `aria-describedby` on Home Screen only.

**Comment by terracoda** [comment]
@zepumph, like you already noticed, the logic that is handling the screen button names and the mini screen descriptions might need some attention.

The screen button names on first screen are not right in either RaP or CL.
![Screen Shot 2020-07-16 at 12 48 36 PM](https://user-images.githubusercontent.com/1902116/87689961-6a758d80-c763-11ea-9068-65a874f1fd5b.png)

Here are the screen button names on the Home Screen
![Screen Shot 2020-07-16 at 12 51 16 PM](https://user-images.githubusercontent.com/1902116/87690388-f8ea0f00-c763-11ea-8e69-54c1e0f0b144.png)

**Comment by terracoda** [comment]
I opened https://github.com/phetsims/coulombs-law/issues/119 in CL's repo since RaP does not have this problem.

**Comment by terracoda** [comment]
@zepumph, I think my review is complete for now. A second review may be needed after all items and issues are addressed.

**Comment by terracoda** [comment]
I am un-assigning myself for now. Please let me know if you have any questions.

**Comment by terracoda** [comment]
@zepumph, I removed the "ready-to-review" label because I reviewed it. I am assuming that might be good protocol. Let me know if that is not good protocol.

**Comment by zepumph** [comment]
@terracoda, is everything to implement delegated to separate issues?

**Comment by zepumph** [comment]
Wait nevermind. I see things here to work on. Sorry to bother!

**Comment by zepumph** [comment]
Sorry for any confusion, I believe that I implemented all recommended changes from this issue over in #641. Over there are changes to labels, regions, and headings for the home screen and nav bar. Please see that issue for review. I'm going to close this issue. Please reopen if you disagree, and feel free to make separate issues as more comes up.

**Comment by jessegreenberg** [comment]
In  https://github.com/phetsims/ratio-and-proportion/issues/116  @terracoda said
> "sim screen" alone in the aria-roledescription for the sim screen buttons likely won't work for learners. They will not know it is a button and that they need to press space or enter to choose a screen.

> The aria-roledescription, if we use it, needs to have the word "button" in it.

**Comment by terracoda** [comment]
@jessegreenberg, I see in this issue, I said to remove the `aria-roledecription`. That is also a completely, logical path.

The main thing is we need the word button from the element's native role, or in the `aria-roledescription`.

**Comment by terracoda** [comment]
@jessegreenberg, the word "Screen" was added to the button labels, so best to remove aria-roledescription all together from the sim screen buttons.

**Comment by jessegreenberg** [comment]
In the above commit I just removed the aria-roledescription from the screen buttons on the home screen, screen buttons in the navigation bar, and the home screen button.

@terracoda can you please review?

**Comment by terracoda** [comment]
Sounds perfect now! Thanks @jessegreenberg!
